### ISSUE TITLE
Can't download Instagram videos/reals in an iOS compatiable codec

### ISSUE THREAD

**Issue by Jarboer** [alexta69/metube#469]
For some reason Instagram videos/reals are the only videos I download that aren't compatible with iOS since they download as AV1 instead of h.264. Videos from Twitter work fine for me. 

I've tried using the Quality: Best (iOS) and Format: MP4 options but it doesn't work. I also tried setting the environment variable YTDL_OPTIONS to {"format": "mp4", "quality": "best_ios"} but that also doesn't change anything. 

I'm running this in a TrueNAS app as the "apps" user (GUI = 568). Here's an example reel that downloads in AV1 despite my settings: [https://www.instagram.com/reel/C8MCIaDIJjV/?igsh=MWhodHRwcHQzdHkzeQ==](https://www.instagram.com/reel/C8MCIaDIJjV/?igsh=MWhodHRwcHQzdHkzeQ==).

Any help would be appericated if I'm missing something. And I can provide more details if needed.

**Comment by Jarboer** [comment]
Here's my logs, it does show the quality is best_ios

[metube_metube-6dd9884f46-4l49g_metube.log](https://github.com/user-attachments/files/16200822/metube_metube-6dd9884f46-4l49g_metube.log)

**Comment by PikuZheng** [comment]
I don't think `best_ios` is a valid value https://github.com/yt-dlp/yt-dlp?tab=readme-ov-file#format-selection

**Comment by Jarboer** [comment]
@PikuZheng Thanks for the reply and link. I believe it should be valid though. It was added here: [https://github.com/alexta69/metube/pull/455](https://github.com/alexta69/metube/pull/455)

Here's some examples of me using the different quality settings:

---

Using these settings:

<img width="962" alt="Screenshot 2024-07-13 at 12 22 05 AM" src="https://github.com/user-attachments/assets/9db8567d-4862-4dcd-af39-6e69d6fa83a9">

This is the relevant logs:

2024-07-13 00:20:04.871689-04:00INFO:ytdl:adding https://www.instagram.com/reel/C8MCIaDIJjV/?igsh=MWhodHRwcHQzdHkzeQ==: quality='best_ios' format='mp4' already=None folder=None custom_name_prefix=''
2024-07-13 00:20:07.732477-04:00INFO:ytdl:downloading Video by j.o.herbert

---

If I use these settings:

<img width="966" alt="Screenshot 2024-07-13 at 12 23 08 AM" src="https://github.com/user-attachments/assets/7dcdd816-8753-4d86-9a7c-95e56f84694a">

I get this:

2024-07-13 00:23:10.197243-04:00INFO:ytdl:adding https://www.instagram.com/reel/C8MCIaDIJjV/?igsh=MWhodHRwcHQzdHkzeQ==: quality='best' format='mp4' already=None folder=None custom_name_prefix=''
2024-07-13 00:23:13.770840-04:00INFO:ytdl:downloading Video by j.o.herbert

**Comment by PikuZheng** [comment]
I never used it. Sorry.
However, I also faced the problem of the annoying vp9 encoding format. I chose to make av01 the preferred format by setting `"format_sort":["codec:avc:m4a"]` in `YTDL_OPTIONS`. It works fine with youtube.
But wait... Does Instagram provide videos in the corresponding format?

**Comment by PikuZheng** [comment]
```
/ # yt-dlp --list-formats https://www.instagram.com/reel/C8MCIaDIJjV/?igsh=MWhodHRwcHQzdHkzeQ== --cookies /downloads/.metube/cookies.txt
[Instagram] Extracting URL: https://www.instagram.com/reel/C8MCIaDIJjV/?igsh=MWhodHRwcHQzdHkzeQ==
[Instagram] C8MCIaDIJjV: Downloading video info
[info] Available formats for C8MCIaDIJjV:
ID                      EXT RESOLUTION │   FILESIZE  TBR PROTO │ VCODEC         VBR ACODEC     ABR ASR MORE INFO
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
dash-794956116038447ad  m4a audio only │ ≈ 72.15KiB  56k https │ audio only         mp4a.40.5  56k 44k DASH audio, m4a_dash
385501297831617v-0      mp4 540x960    │                 https │ unknown            unknown
385501297831617v-1      mp4 540x960    │                 https │ unknown            unknown
dash-1661688841233207vd mp4 720x1280   │ ≈276.61KiB 214k https │ vp09.00.31.08 214k video only         DASH video, mp4_dash
443611441717854v        mp4 1080x1920  │                 https │ unknown            unknown
dash-1887139295033561v  mp4 1080x1920  │ ≈395.29KiB 306k https │ vp09.00.40.08 306k video only         DASH video, mp4_dash
```
The "unknown" is actually AVC1, which may be because Instagram did not provide the correct information.
So yt-dlp will not select it, but only vp09. https://github.com/yt-dlp/yt-dlp/issues/4390 fyi

**Comment by alexta69** [comment]
Is this still a problem with the latest release @Jarboer ?

**Comment by Jarboer** [comment]
@alexta69 Yes it is still an issue. The video won't play on iOS (iPhone 12 on iOS 18 RC) in Safari. Weirdly though I noticed, while the video won't play in the Files app, the audio still plays despite there being no video or play back controls.

**Comment by Jarboer** [comment]
@alexta69 Wait I just realized the version I updated to in TrueNAS was not the latest as it isn't avaliable yet (08-22 rather than 9-11). I'll try again once it is. Hopefully, it does work, but we'll have to see if it does! Sorry for the extra ping.

**Comment by Jarboer** [comment]
@alexta69 I was able to update to the actual latest version and it is still an issue unfortunately.

**Comment by proof001** [comment]
Did you

> @alexta69 I was able to update to the actual latest version and it is still an issue unfortunately.

Have you found a solution for this?

**Comment by sarangtc** [comment]
> I never used it. Sorry. However, I also faced the problem of the annoying vp9 encoding format. I chose to make av01 the preferred format by setting `"format_sort":["codec:avc:m4a"]` in `YTDL_OPTIONS`. It works fine with youtube. But wait... Does Instagram provide videos in the corresponding format?

this gives error:
Error adding URL: ERROR: Invalid format sort string ":" given by extractor; please report this issue on https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using yt-dlp -U

yt-dlp
2025.06.30
MeTube
2025-07-01

**Comment by PikuZheng** [comment]
>Invalid format sort string ":"

means only a ":" in format_sort value

**Comment by NoroNetwork** [comment]
Hello, any update on this? Youtube works great. But instagram is downloaded in VP9 codec even when H264 is selected. 
I can play it in VLC but not natyve mac suport for that. 

<img width="515" height="493" alt="Image" src="https://github.com/user-attachments/assets/ab701b3f-ae61-4fec-9d59-8ed116e261ad" />

**Comment by alexta69** [comment]
Root cause is Instagram not exposing codec info, so yt-dlp can't select h264 (yt-dlp/yt-dlp#4390). Nothing MeTube can do here. Closing.
